### PR TITLE
qat: heartbeat support + new capabilities 

### LINKED ISSUE #1476: qat: add support for the new services configurations on qat_4xxx
https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-driver-qat

The `cfg_services` supports more fields, essentially:
```
What:		/sys/bus/pci/devices/<BDF>/qat/cfg_services
Date:		June 2022
KernelVersion:	6.0
Contact:	qat-linux@intel.com
Description:	(RW) Reports the current configuration of the QAT device.
		Write to the file to change the configured services.

		The values are:

		* sym;asym: the device is configured for running crypto
		  services
		* asym;sym: identical to sym;asym
		* dc: the device is configured for running compression services
		* sym: the device is configured for running symmetric crypto
		  services
		* asym: the device is configured for running asymmetric crypto
		  services
		* asym;dc: the device is configured for running asymmetric
		  crypto services and compression services
		* dc;asym: identical to asym;dc
		* sym;dc: the device is configured for running symmetric crypto
		  services and compression services
		* dc;sym: identical to sym;dc

```

Add support for these and think about the new `qat.intel.com/<resource>` names.

### LINKED ISSUE #1480: qat: add support for device heartbeat status
https://git.kernel.org/pub/scm/linux/kernel/git/herbert/cryptodev-2.6.git/commit/?id=359b84f8db942ef46d24de8aa397790c3fae22e0

### PR DESCRIPTION
**Pull request by tkatila**
Fixes #1480
Fixes #1476

### CODE REVIEW

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go [hunk 1]
```diff
@@ -358,6 +358,22 @@ func (dp *DevicePlugin) getDpdkMounts(dpdkDeviceName string) []pluginapi.Mount {
 	}
 }
 
+// heartbeat is only supported on the 4xxx devices. Other devices return "unknown".
```
Comment (mythi) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv.go:361]: I believe it's available on `c62x, c3xxx,dh895xcc` too? the difference was that on `4xxx` the poll time was fixed.

Comment (tkatila) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv.go:361]: I think you are right. For the `heartbeat/config` it says "This parameter is effective only for c3xxx, c62x, dh895xcc devices."

I'll modify the flow.

**Comment by tkatila** [comment]
Tested with kernel 6.6 rc5 and 4xxx qat device.

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go [hunk 2]
```diff
@@ -426,6 +455,14 @@ func getDeviceCapabilities(device string) (string, error) {
...
 	case "asym":
 		return "asym", nil
+	case "asym;dc":
+		return "asym-dc", nil
+	case "dc;asym":
+		return "asym-dc", nil
+	case "sym;dc":
+		return "sym-dc", nil
+	case "dc;sym":
+		return "sym-dc", nil
```
Comment (mythi) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv.go:465]: The naming of the new resources looks reasonable. The question is do we harmonize `asym-sym` with the new naming or keep `cy`. I believe the latter is better to not cause any breakages ...

We have documentation for these in the top-level README and qat_plugin/README. Can you make the necessary updates to those as well (including the updated permalink to the sysfs abi).

File: demo/qat-init.sh [hunk 3]
```diff
@@ -3,7 +3,7 @@
 NODE_NAME="${NODE_NAME:-}"
 ENABLED_QAT_PF_PCIIDS=${ENABLED_QAT_PF_PCIIDS:-37c8 4940 4942 4944}
 DEVS=$(for pf in $ENABLED_QAT_PF_PCIIDS; do lspci -n | grep -e "$pf" | grep -o -e "^\\S*"; done)
-SERVICES_LIST="sym;asym dc"
+SERVICES_LIST="sym asym sym;asym dc sym;dc asym;dc dcc"
```
Comment (mythi) [inline demo/qat-init.sh:6]: `dcc` does not exist yet so let's leave it out for now:

```suggestion
SERVICES_LIST="sym asym sym;asym dc sym;dc asym"
```

File: cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go [hunk 4]
```diff
@@ -552,6 +666,7 @@ func TestScan(t *testing.T) {
 			if !tt.expectedErr && err != nil {
 				t.Errorf("got unexpected error: %+v", err)
 			}
+
```
Comment (mythi) [inline cmd/qat_plugin/dpdkdrv/dpdkdrv_test.go:669]: nit: drop?

**Review by mythi — CHANGES_REQUESTED** [review]

**Comment by mythi** [comment]
+ add `Fixes: ` tags

File: demo/qat-init.sh (hunk 3 above)
Comment (tkatila) [reply demo/qat-init.sh:6]: Sure. Your suggestion would remove too much, though.

File: cmd/qat_plugin/dpdkdrv/dpdkdrv.go (hunk 2 above)
Comment (tkatila) [reply cmd/qat_plugin/dpdkdrv/dpdkdrv.go:465]: Will do.

File: demo/qat-init.sh (hunk 3 above)
Comment (mythi) [reply demo/qat-init.sh:6]: oops, yes. updated

**Review by mythi — APPROVED** [review]

**Pull request merged**